clang/lib/Sema/SemaStmtAttr.cpp

```diff
@@ -322,6 +322,90 @@ static Attr *handleUnlikely(Sema &S, Stmt *St, const ParsedAttr &A,
   return ::new (S.Context) UnlikelyAttr(S.Context, A);
 }
 
+CodeAlignAttr *Sema::BuildCodeAlignAttr(const AttributeCommonInfo &CI,
+                                        Expr *E) {
+  if (!E->isValueDependent()) {
+    llvm::APSInt ArgVal;
+    ExprResult Res = VerifyIntegerConstantExpression(E, &ArgVal);
+    if (Res.isInvalid())
+      return nullptr;
+    E = Res.get();
+
+    // This attribute requires an integer argument which is a constant power of
+    // two between 1 and 4096 inclusive.
+    // Check if an expression resulting in a negative/positive cases that take
+    // more than 64 bits.
+    // Print the actual alignment value passing to the attribute if it is less
+    // than 64 bits otherwise print the expression.
+    if (ArgVal < CodeAlignAttr::MinimumAlignment ||
+        (ArgVal < CodeAlignAttr::MaximumAlignment && !ArgVal.isPowerOf2())) {
+      if (!ArgVal.trySExtValue()) {
```

Comment:
```suggestion
      if (std::optional<int64_t> Value = ArgVal.trySExtValue()) {
```

Invert the cases here so you can use the value that you got out of the `trySExtValue`.

Reply:
@erichkeane, I tried your suggested change. But it is casing the assertion. Am i missing something? 
```
if (ArgVal < CodeAlignAttr::MinimumAlignment ||
        (ArgVal < CodeAlignAttr::MaximumAlignment && !ArgVal.isPowerOf2())) {
      if (std::optional<int64_t> Value = ArgVal.trySExtValue())
        Diag(CI.getLoc(), diag::err_attribute_power_of_two_in_range)
            << CI << CodeAlignAttr::MinimumAlignment
            << CodeAlignAttr::MaximumAlignment << E;
      else
        Diag(CI.getLoc(), diag::err_attribute_power_of_two_in_range)
            << CI << CodeAlignAttr::MinimumAlignment
            << CodeAlignAttr::MaximumAlignment << ArgVal.getSExtValue();
      return nullptr;
    }

    if (ArgVal > CodeAlignAttr::MaximumAlignment) {
      if (std::optional<int64_t> Value = ArgVal.trySExtValue())
        Diag(CI.getLoc(), diag::err_attribute_power_of_two_in_range)
            << CI << CodeAlignAttr::MinimumAlignment
            << CodeAlignAttr::MaximumAlignment << E;
      else
        Diag(CI.getLoc(), diag::err_attribute_power_of_two_in_range)
            << CI << CodeAlignAttr::MinimumAlignment
            << CodeAlignAttr::MaximumAlignment << ArgVal.getZExtValue();
      return nullptr;
    }
```

Reply:
Yes, you have to change `ArgVal.getSExtValue` in the `else` of each one back to the Expression to print instead.

Reply:
Thanks!.
Done